services/src/main/java/org/keycloak/broker/saml/SAMLEndpoint.java

```diff
@@ -433,11 +438,25 @@ protected Response handleLoginResponse(String samlResponse, SAMLDocumentHolder h
                     return ErrorPage.error(session, authSession, Response.Status.BAD_REQUEST, Messages.INVALID_REQUESTER);
                 }
 
-                Element assertionElement;
+                List<KeyWrapper> encRsaKeys = session.keys()
```

Comment:
Does it make more sense to only query for enc keys if the assertion is encrypted?

Reply:
Nice find! Yes, that should be queried only if it's required! I'll change that

Reply:
@pedroigor It should be ok now. 
It loads fetches the key only if one or both the assertion and/or the key are encrypted ;)
Let's resume the PR review!